libvast/src/system/index.cpp

```diff
@@ -1035,6 +1035,12 @@ index(index_actor::stateful_pointer<index_state> self,
         .then(
           [=, candidates
               = std::move(candidates)](meta_index_result& midx_result) mutable {
+            if (auto it = self->state.monitored_queries.find(sender->address());
```

Comment:
After re-reading the commit I'm not so sure anymore that `sender` always refers to the correct actor here, at least for jobs that are taken from the backlog we probably need to pass an explicit `sender` argument to this actor.

Reply:
I've added another commit to make this more robust.